common/lib/mysql_client_wrapper.ts

```diff
@@ -34,15 +37,21 @@ export class MySQLClientWrapper implements ClientWrapper {
    * @param targetClient The community driver client created for an instance.
    * @param hostInfo Host information for the connected instance.
    * @param properties Connection properties for the target client.
+   * @param driverDialect The driver dialect to obtain driver-specific information.
    */
-  constructor(targetClient: any, hostInfo: HostInfo, properties: Map<string, any>) {
+  constructor(targetClient: any, hostInfo: HostInfo, properties: Map<string, any>, driverDialect: DriverDialect) {
     this.client = targetClient;
     this.hostInfo = hostInfo;
     this.properties = properties;
+    this.driverDialect = driverDialect;
     this.id = uniqueId("MySQLClient_");
   }
 
   query(sql: any): Promise<any> {
+    const queryTimeout = this.properties.get(WrapperProperties.WRAPPER_QUERY_TIMEOUT.name);
+    if (queryTimeout && !sql[this.driverDialect.queryTimeoutPropertyName]) {
+      sql[this.driverDialect.queryTimeoutPropertyName] = queryTimeout;
```

Comment:
This line is very driver specific, it assumes the sql is a obj instead of a string so I think this logic belongs in driver dialect as well.
What do you think of adding a static method
```ts
  static prepareQuery(sql: any, properties: Map<string, any>) {
    const queryTimeout = properties.get(WrapperProperties.WRAPPER_QUERY_TIMEOUT.name);
    if (queryTimeout && !sql[MySQL2DriverDialect.queryTimeoutPropertyName]) {
      sql[MySQL2DriverDialect.queryTimeoutPropertyName] = queryTimeout;
    }
    return sql;
  }
```

This way this method can be simplified and you won't need to pass driver dialects around anymore.
```
  query(sql: any): Promise<any> {
    return this.client?.query(DriverDialect.prepareQuery(sql, this.properties));
  }
```

Reply:
@sergiyvamz

Reply:
I partially agree here. I think this piece of code is very driver specific and it should be moved to driver dialect class. And I don't agree with @karenc-bq about static method. That makes a static bond between client and dialect. It'd cause issues mocking and testing. Why not to make a `getQueryTimeout()` method in driver dialect class? The method can do type conversion if needed.

Reply:
If I understood you correctly, the `getQueryTimeout()` method will be used like this for mysql
```ts
return this.client?.query(this.driverDialect.getQueryTimeout(sql, this.properties));
```
and like this for pg?
```
this.properties = this.driverDialect.getQueryTimeout(sql, this.properties);
return this.client?.queryWithTimeout(sql, this.properties);
```